src/SingleLinkedList.h

```diff
@@ -226,6 +227,7 @@ T* SingleLinkedList<T>::removeLast() {
 	if(size ==1) {
 		tail = head = NULL;
 		size--;
+		delete currentNode;
 		return currentNode->data;
```

Comment:
Since you deleted Current Node you should no longer be able to access data. Thus you need to save the data before hand then delete the node and then return the data.

```c++
tail = head = NULL;
size--;
T* placeHolder = currentNode->data;
delete currentNode;
return placeHolder;
```

I think thats what we want.